content/posts/mirrorbits-cdn.md

```diff
@@ -0,0 +1,390 @@
+    #
+    location ~ \.php$ {
+        fastcgi_split_path_info ^(.+?\.php)(/.*)$;
+        fastcgi_param HTTP_PROXY "";
+        fastcgi_pass unix:/run/php/php7.3-fpm.sock;
+        fastcgi_index index.php;
+        include fastcgi_params;
+        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+    }
+}
+```
+
+Some of the configuration here is worthy of describing in detail.
+
+The Kodi selector is used due to the Kodi backend transition from Python 2 to Python 3. Thus, some versions require assets for Python 2, and newer versions for Python 3. This selector operates based on the sent user agent of the client, deciding which version of Kodi they are using, and thus directing them to the right location.
+
+The main locations are `/archive`, `/releases`, and `/master`. The first contains archives and is not forwarded on to the Mirrorbits process (despite the files being synced above) due to the `try_files` directive. The second is the main repository directory, and the third is a duplicate location that is used for fallback and also does not redirect to Mirrorbits.
+
+Next is the main Mirrorbits handler. The forwarding is based off the file extension of the requested file. Thus, when loading, e.g. the PHP index pages, the requests are not forwarded; only requests for the listed file types are forwarded on to the Mirrorbits process to be distributed to mirrors.
+
+The next 3 options are for Mirrorbits status pages, which provide information on the currently available mirrors. For any file (e.g. https://repo.jellyfin.org/releases/server/debian/stable/meta/jellyfin_10.7.2-1_all.deb), one can append the `/mirrorlist` or `/mirrorinfo` locations to show information about the available mirrors. Try it yourself: https://repo.jellyfin.org/releases/server/debian/stable/meta/jellyfin_10.7.2-1_all.deb/mirrorlist. Finally the `/mirrorstats` page, whether on a file or at the root of the domain (https://repo.jellyfin.org/mirrorstats) shows the current status of the mirrors in general, including if any are offline.
+
+All together, these NGiNX configs provide the foundation for Mirrorbits to work, and this was the part that actually took the longest. Thanks to [@PalinuroSec](https://github.com/PalinuroSec) on GitHub for his [fantastic example gist](https://gist.github.com/PalinuroSec/f0bfb815240573ab1b0b58f3c76620d4).
+
+## Mirrorbits: The Workhorse
+
+Installing Mirrorbits is quite straightforward: I simply downloaded the latest binary release (`0.5.1`) from the Mirrorbits repository, installed the `mirrorbits` binary to `/usr/local/bin`, the sample configuration to `/etc/mirrorbits.conf`, and the templates to `/usr/local/share/mirrorbits`. Unfortunately, Mirrorbits development seems to have stalled since 2018, including documentation. [I opened an issue](https://github.com/etix/mirrorbits/issues/105) during my troubleshooting that is still unanswered, which is unfortunate, and I hope this blog post to be able to resolve that issue.
```

Comment:
```suggestion
Installing Mirrorbits is quite straightforward: I simply downloaded the latest binary release (`0.5.1`) from the Mirrorbits repository, installed the `mirrorbits` binary to `/usr/local/bin`, the sample configuration to `/etc/mirrorbits.conf`, and the templates to `/usr/local/share/mirrorbits`. Unfortunately, Mirrorbits development seems to have stalled since 2018, including documentation. [I opened an issue](https://github.com/etix/mirrorbits/issues/105) during my troubleshooting that is still unanswered, which is unfortunate, and I hope this blog post is able to resolve that issue.
```

Not sure exactly what you mean here though

Reply:
"will be" is the proper thing I was going for, my flowery language getting the better of me here.